metadata-ingestion/tests/unit/stateful_ingestion/provider/test_provider.py

```diff
@@ -181,3 +185,32 @@ def test_providers(self):
                 state_class=type(job2_state_obj),
             )
             self.assertEqual(job2_last_checkpoint, job2_checkpoint)
+
+    def test_state_provider_wrapper(self):
+        ctx: PipelineContext = PipelineContext(
+            run_id=self.run_id, pipeline_name=self.pipeline_name
+        )
+        ctx.graph = self.mock_graph
+        # Test 1: stateful_ingestion_config provided with enabled as true
+        state_provider = StateProviderWrapper(
+            StatefulIngestionConfig(enabled=True), ctx
+        )
+        assert state_provider.stateful_ingestion_config
+        assert state_provider.ingestion_checkpointing_state_provider
+        ctx.checkpointers = {}
```

Comment:
the reuse of ctx is pretty confusing - can you split this up into 4 different tests (eg 4 different `def test_...` methods)